src/main/python/image_preprocessing/build_director.py

```diff
@@ -17,45 +17,65 @@ class BuildDirector:
     The BuildDirector constructs the Pipeline using the PipelineBuilder
     """
     @staticmethod
-    def construct(preferences):
+    def construct(preferences, identification_type):
         """
         This function constructs the pipeline
         Author(s):
-            Nicolai van Niekerk
+            Nicolai van Niekerk and Marno Hermann
         Args:
             preferences (dict): User-specified techniques to use in pipeline.
+            identification_type (string): Containts the type of identification, this is used
+                                        to determine which techniques are used.
         Returns:
             :Pipeline (Constructed pipeline)
         """
         builder = PipelineBuilder()
-
         # Use template matching to identify type here
 
         if 'blur_method' in preferences:
             blur_method = preferences['blur_method']
+        elif identification_type['type'] == 'idcard':
+            blur_method = 'gaussian'
+        elif identification_type['type'] == 'idbook':
+            blur_method = 'median'
+        elif identification_type['type'] == 'studentcard':
+            blur_method = 'median'
         else:
-            """
-                if type == 'card':
-                    blur_method = 'gaussian'
-                else if type == 'book':
-                    blur_method = 'median'
-            """
+            # Default
             blur_method = 'median'
 
         if blur_method == 'median':
             blur_kernel_size = [9]
         else:
-            blur_kernel_size = [(3, 3)]
+            if identification_type['type'] == 'idbook':
+                blur_kernel_size = [(7, 7)]
+            else:
+                blur_kernel_size = [(3, 3)]
 
         if 'threshold_method' in preferences:
             threshold_method = preferences['threshold_method']
+        elif identification_type['type'] == 'idcard':
+            threshold_method = 'otsu'
+        elif identification_type['type'] == 'idbook':
+            threshold_method = 'adaptive'
+        elif identification_type['type'] == 'studentcard':
+            threshold_method = 'adaptive'
         else:
             # Default
             threshold_method = 'adaptive'
 
         if 'color' in preferences:
             color_extraction_type = 'extract'
             color = preferences['color']
+        elif identification_type['type'] == 'idcard':
+            color_extraction_type = 'extract'
+            color = 'red'
+        elif identification_type['type'] == 'idbook':
+            color_extraction_type = 'extract'
+            color = 'red'
+        elif identification_type['type'] == 'studentcard':
```

Comment:
If `identification_type` was a string this would've broken. Why is `identification_type` still a dict?